sticker/src/depparse/post_processing.rs

```diff
@@ -29,6 +31,48 @@ where
     }
 }
 
+/// Break cycles in the graph.
+///
+/// Panics when a token does not have a head.
+pub fn break_cycles(sent: &mut Sentence, root_idx: usize) {
+    loop {
+        let components = {
+            let digraph: &DiGraph<_, _> = (&*sent).into();
+            tarjan_scc(digraph)
+                .into_iter()
+                .filter(|c| c.len() > 1)
+                .collect::<Vec<_>>()
+        };
+
+        // We are done if there are no more cycles.
+        if components.is_empty() {
+            break;
+        }
+
+        for cycle in components.into_iter().filter(|c| c.len() > 1) {
+            // Find the first token in the cycle, exclude the root
+            // token to avoid self-cycles.
+            let first_token = cycle
+                .into_iter()
+                .filter(|idx| idx.index() != root_idx)
+                .min()
+                .expect("Cannot get minimum from a non-empty iterator")
```

Comment:
"from an empty iterator"?

Reply:
I did mean non-empty, because that is what the iterator is guaranteed to be at this point. Changed the error message a bit.

Reply:
The rustdoc of `.min()` states that the method returns `None` if the iterator is empty. The iterator is guaranteed to be non-empty, but I guess, `.expect()` or `.unwrap()` don't make much of a difference here. I just didn't correctly parse the string argument.